src/main/kotlin/com/openlattice/graph/partioning/RepartitioningJob.kt

```diff
@@ -375,71 +383,54 @@ INSERT INTO ${E.name} SELECT $REPARTITION_EDGES_COLUMNS
 /**
  * Computes the actual partition and compares it to current partition. If partitions do not match deletes the row.
  * 1. entity set id
- * 2. partitions (array)
- * 3. partition
- * 4. entity set id
+ * 2. partition
+ * 3. partitions
+ * 4. array length ( partitions )
+ * 5. array length ( partitions )
+ * 6. array length ( partitions )
  *
  */
 private val DELETE_DATA_SQL = """
 DELETE FROM ${DATA.name} 
-    USING (SELECT ${ID.name},${ENTITY_SET_ID.name},${PARTITION.name},${PARTITIONS.name} FROM ${DATA.name} INNER JOIN (select ? as ${ENTITY_SET_ID.name},? as ${PARTITIONS.name} ) as es USING (${ENTITY_SET_ID.name})) as to_be_deleted
-    WHERE 
-      ${DATA.name}.${PARTITION.name} = ?
-      AND ${DATA.name}.${ENTITY_SET_ID.name} = ?
-      AND ${DATA.name}.${PARTITION.name}!=${getPartitioningSelector(DATA.name + "." + ID.name)} 
-      AND to_be_deleted.${ID.name} = ${DATA.name}.${ID.name} 
-      AND to_be_deleted.${PARTITION.name} = ${DATA.name}.${PARTITION.name};  
+    WHERE
+      ${DATA.name}.${ENTITY_SET_ID.name} = ?
+      AND ${DATA.name}.${PARTITION.name} = ?
```

Comment:
Any reason not to do this as one big batch delete, ie `WHERE partition = ANY(allOldPartitions)`

Reply:
It processes one partition at a time so that it can resume on the current partition if the job is paused or the node crashes.